### PR TITLE
[SYCL][NewPM] Disable vectorization and loop transformation

### PR DESCRIPTION
**Pull request by mlychkov**
Port changes that disable vectorization and loop transformation passes
in optimization pipeline from legacy Pass Manager builder to new PM builder
(see commit ff6929e6317ccd6dbc313e5e0020a375148564f5).
Also for those tests that fail without this fix after optimization by
the new Pass Manager add validation for both legacy and new PMs.

Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

**Comment by mlychkov** [comment]
The full set of fixes for new PM also validated in #4860.

**Review by bader — DISMISSED** [review]

File: sycl/test/esimd/vadd.cpp [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -o %t.out
 // RUN: %RUN_ON_HOST %t.out
 
-// Check that the code compiles with -O0 and -g
-// RUN: %clangxx -I %sycl_include %s -o %t.out -fsycl -O0
-// RUN: %clangxx -I %sycl_include %s -o %t.out -fsycl -O0 -g
+// Check that the code compiles with -O0 and -g on both legacy and new Pass
+// Managers
+// RUN: %clangxx -I %sycl_include %s -o %t.out -fsycl -fno-legacy-pass-manager -O0
```
Comment (kbobrovs) [inline sycl/test/esimd/vadd.cpp:9]: why this is added to esimd tests? How these 3 tests were selected for such modification?

**Review by kbobrovs — CHANGES_REQUESTED** [review]

File: sycl/test/esimd/vadd.cpp (hunk 1 above)
Comment (mlychkov) [reply sycl/test/esimd/vadd.cpp:9]: They crash on SYCLLowerESIMD pass execution step if loop transformation and vectorization passes are enabled in CodeGen.

**Comment by AlexeySachkov** [comment]
@mlychkov, could you please resolve merge conflicts?

File: sycl/test/esimd/vadd.cpp (hunk 1 above)
Comment (mlychkov) [reply sycl/test/esimd/vadd.cpp:9]: @kbobrovs Could you please provide feedback or approve the PR if there are no more concerns?

Comment (kbobrovs) [reply sycl/test/esimd/vadd.cpp:9]: I see. @mlychkov - could you please instead mark them as XFAIL, add a TODO and file an internal ticket?

Comment (AlexeySachkov) [reply sycl/test/esimd/vadd.cpp:9]: @kbobrovs, the point is that those tests fail with new pass manager _without_ changes proposed by @mlychkov, but they do pass with this PR using both pass managers

Comment (kbobrovs) [reply sycl/test/esimd/vadd.cpp:9]: Ok, I think I get it now. Thanks!

**Review by kbobrovs — APPROVED** [review]

**Comment by mlychkov** [comment]
Precommit tests failed due to infrastructure issue.

**Comment by bader** [comment]
@intel/dpcpp-tools-reviewers, ping.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by mlychkov** [comment]
The issue about failed pre-commit on Windows has been raised.

**Pull request merged**